trie-db/src/iterator.rs

```diff
@@ -77,30 +76,31 @@ impl<'a, L: TrieLayout> TrieDBNodeIterator<'a, L> {
 			trail: Vec::with_capacity(8),
 			key_nibbles: NibbleVec::new(),
 		};
-		db.root_data().and_then(|root_data| {
-			r.descend(&root_data, Some(db.root().clone())).map(|_| ())
-		})?;
+		let (root_node, root_hash) = db.get_raw_or_lookup(db.root().as_ref(), EMPTY_PREFIX)?;
```

Comment:
- root_data become root function, there is only one place root_data seems still use: maybe lets remove the function or keep using it but not a middle ground.
(iterator.rs line 206).